Networking Guide

hramos opened the pull request:
Simplified Networking Guide, based on the old Network polyfill doc.

This guide strongly recommends using fetch, while still informing the user about React Native's support for other libraries.

In order to provide an actual working networking example, a `movies.json` file is added at the root of the site, allowing the user to fetch a small blob of JSON:

```
fetch('http://facebook.github.io/react-native/movies.json')
```

![networking](https://cloud.githubusercontent.com/assets/165856/16321804/d2bd7c6a-3953-11e6-9fc5-30baaa38d7a4.png)

facebook-github-bot-2 commented:
By analyzing the blame information on this pull request, we identified @grgmo and @alexkrolick to be potential reviewers.

lacker commented:
Your paragraph on websocket seems a bit redundant. Do you think it would be worth adding a little bit of sample code to it, or does that link explain things sufficiently?

lacker commented on docs/Basics-Network.md:
```diff
@@ -1,25 +1,27 @@
 ---
 id: basics-network
-title: Network
+title: Networking
```
+1

lacker commented on docs/Basics-Network.md:
```diff
@@ -1,25 +1,27 @@
 ---
 
-One of React Native's goals is to be a playground where we can experiment with different architectures and crazy ideas. Since browsers are not flexible enough, we had no choice but to reimplement the entire stack. In the places that we did not intend to change anything, we tried to be as faithful as possible to the browser APIs. The networking stack is a great example.
+Many modern mobile apps will at some point find the need to load resources from a remote endpoint. You may want to make a POST request to a REST API, or you may simply need to fetch a chunk of static content from another server.
```
remove "will at some point find the"

lacker replied:
also remove "modern"

lacker commented on docs/Basics-Network.md:
```diff
@@ -1,25 +1,27 @@
-#### Usage
+Fetch will seem familiar if you have used `XMLHttpRequest` or other networking APIs before. You may refer to MDN's guide on [Using Fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) for additional information.
+
+In order to fetch content from an arbitrary endpoint, just pass the URL to fetch:
```
let's call it "URL" rather than "endpoint" in these docs

hramos commented:
The second, redundant sentence was unintentional. Just removed it. 

The original doc did have some sample code for WebSocket that I removed. I think it makes sense to add back at least one of those code blocks, and link elsewhere if the reader wants to learn more.

lacker commented on docs/Basics-Network.md:
````diff
@@ -35,121 +37,47 @@ fetch('https://mywebsite.com/endpoint/', {
-```
+## WebSocket Support
 
+React Native supports [WebSocket](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket), a protocol which provides full-duplex communication channels over a single TCP connection.
````
can we have WebSocket somewhere within our own autogenerated docs? Then you could link to that. I'm suspicious that the mozilla.org link defines a 95%-the-same-but-5%-different thing.

hramos commented:
Alright, reworking things a bit and added back some code samples.

lacker commented:
@facebook-github-bot shipit
